src/XMakeTasks/CodeTaskFactory.cs

```diff
@@ -771,6 +771,16 @@ private Assembly CompileInMemoryAssembly()
                 if (!s_compiledTaskCache.TryGetValue(fullSpec, out existingAssembly))
                 {
                     // Invokes compilation. 
+
+                    // Note: CompileAssemblyFromSource uses Path.GetTempPath() directory, but will not create it. In some cases 
+                    // this will throw inside CompileAssemblyFromSource. To work around this, ensure the temp directory exists. 
+                    // See: https://github.com/Microsoft/msbuild/issues/328
+                    var tempPath = Path.GetTempPath();
+                    if (!Directory.Exists(tempPath))
```

Comment:
I don't feel strongly about it.  The usual argument is that you introduce a timing issue between the check and the action, but that doesn't seem to be a big problem here because there's a hole between directory and file creation anyway.